### DISCUSSION TITLE
Error while reading the database: Invalid inner header id size

### DISCUSSION THREAD

**Discussion by blynn357** [keepassxreboot/keepassxc#10332] [Q&A]
I have KeePassXC installed on my Mac mini running the latest version of Mac OS Sonoma. KeePassXC was working fine yesterday. This morning I tried to use it and received this message, "Error while reading the database: Invalid inner header id size". 

A little more background. The database for KeePassXC is stored on a partition on my Mac that if formatted as ExFAT and encrypted. I do this because I also boot into Linux Mint from an external drive attached to this computer. So to be clear both OS's are accessing the database on my Mac mini's internal hard disk. I have accessed this data in the past from both Linux Mint and Mac OS Sonoma.

This would not normally be an issue but I was unable to find a way import my data from 1Password or Bitwarden. So I have spent hours upon hours manually adding my data which is over 1,000 entries into KeePassXC. I have searched for solutions that discuss repairing the database from KeePassXC but I am not finding these options in KeePassXC 2.7.6.

**Comment by droidmonkey** [comment]
Your database file is corrupt and there is nothing that can be done. You will have to restore from a backup. We have an importer for bitwarden and 1password, but it's not merged to develop branch yet. 

https://github.com/keepassxreboot/keepassxc/pull/9815

**Reply by blynn357** [reply]
Thank you Droid Monkey. I did some research to find out where the backups are stored. The luck of the Irish was with me on this. Apple's Time Machine stopped backing up the aforementioned Shared partition and I had not selected to backup in Application Settings/General/File Management.

Do you have any idea when the importers will be merged to the develop branch or can you advise me the best way to check for this 'merger'?

**Reply by blynn357** [reply]
Droid Monkey I was 'saved' by Backblaze. I was under the impression that my Backblaze subscription only saved the previous day's data. In my case it at least went back to Friday which is fine because that means that I only lost Saturday's inputs. I am sharing this because perhaps other users of KeePassXC might have the same problem. 

I also went into Settings and set a backup location for KeePassXC data files to a file on my Mac mini which will be backed up daily by Time Machine. Is there is a place in Settings to set the number of database backups that will be saved?  The closest thing I could find is "Remember previously used databases".
